engine/execution/messages.go

```diff
@@ -11,38 +11,122 @@ import (
 // TODO(patrick): rm unaccessed fields
 type ComputationResult struct {
 	*entity.ExecutableBlock
+	*ExecutionResults
+	*AttestationResults
+
+	*flow.ExecutionReceipt
+}
+
+func NewEmptyComputationResult(
+	block *entity.ExecutableBlock,
+) *ComputationResult {
+	return &ComputationResult{
+		ExecutableBlock:    block,
+		ExecutionResults:   NewEmptyExecutionResults(block),
+		AttestationResults: NewEmptyAttestationResults(block),
+	}
+}
+
+func (cr *ComputationResult) CollectionResult(colIndex int) *ColResSnapshot {
+	if colIndex < 0 && colIndex > cr.ExecutedColCounter {
+		return nil
+	}
+	return &ColResSnapshot{
+		blockHeader: cr.Block.Header,
+		collection: &flow.Collection{
+			Transactions: cr.CollectionAt(colIndex).Transactions,
+		},
+		collectionIndex:        colIndex,
+		isSystemCollection:     colIndex == len(cr.CompleteCollections),
+		updatedRegisters:       cr.StateSnapshots[colIndex].UpdatedRegisters(),
+		readRegisterIDs:        cr.StateSnapshots[colIndex].ReadRegisterIDs(),
+		emittedEvents:          cr.Events[colIndex],
+		transactionResults:     cr.transactionResultsByCollectionIndex(colIndex),
+		executionSnapshot:      cr.StateSnapshots[colIndex],
+		serviceEvents:          cr.ServiceEvents[colIndex],
+		convertedServiceEvents: cr.ConvertedServiceEvents[colIndex],
+	}
+}
+
+// ExecutionResults captures artifacts of execution of block collections
+type ExecutionResults struct {
+	ExecutedColCounter     int
 	StateSnapshots         []*state.ExecutionSnapshot
 	Events                 []flow.EventsList
-	EventsHashes           []flow.Identifier
-	ServiceEvents          flow.EventsList
+	ServiceEvents          []flow.EventsList
+	ConvertedServiceEvents []flow.ServiceEventList
 	TransactionResults     []flow.TransactionResult
```

Comment:
We should probably group the ExecutionResults fields by chunks (and get rid of TransactionResultIndex)

In particular, since you already definedd ColResSnapshot, why not just use it

Reply:
can you merge that into this PR? I suspect a lot of the ColResSnapshot method no longer make sense after merging